packages/flutter/test/widgets/icon_test.dart

```diff
@@ -122,4 +124,22 @@ void main() {
     final RichText richText = tester.firstWidget(find.byType(RichText));
     expect(richText.text.style.fontFamily, equals('Roboto'));
   });
+
+  testWidgets('Icon with semantic label', (WidgetTester tester) async {
+    final SemanticsTester semantics = new SemanticsTester(tester);
+
+    await tester.pumpWidget(
+      const Directionality(
+        textDirection: TextDirection.ltr,
+        child: const Center(
+          child: const Icon(
+              null,
```

Comment:
Can you add a test where this is not null? I think that will expose an extra item in the semantics tree that we don't want there (see previous comment about ExcludeSemantics).